docs/src/main/asciidoc/lra-guide.adoc

```diff
@@ -0,0 +1,196 @@
+`quarkus.lra.coordinator-url=<url>` which specifies the HTTP endpoint of an external
+coordinator, for example:
+
+[source,bash]
+----
+quarkus.lra.coordinator-url=http://localhost:8080/lra-coordinator
+----
+
+For a narayana coordinator the path component of the url is normally `lra-coordinator`.
+Coordinators can be obtained from `https://hub.docker.com/r/jbosstm/lra-coordinator`
+or you can build your own coordinator using a maven pom that includes the appropriate
+dependencies. A quarkus quickstart will be provided to show how to do this or you can
+take a look at one of the https://github.com/jbosstm/quickstart/tree/master/rts/lra-examples/lra-coordinator[narayana quickstarts].
+Another option would be to run it managed inside a WildFly application server.
+
+== Handling failures
+
+When an LRA is told to finish, i.e. when a method annotated with `@LRA(end = true, ...)`
+is invoked, the coordinator will instruct all services involved in the interaction to
+finish. If a service is unavailable (or still finishing) then the coordinator will retry
+periodically. It is the users responsibility to restart failed services on the same
+endpoint that they used when they first joined the LRA, or to tell the coordinator that
+they wish to be notified on new endpoints. An LRA is not deemed finished until *all*
+participants have acknowledged that they have finished.
+
+The coordinator is responsible for reliably creating and ending LRAs and for managing
+participant enlistment and it therefore must be available (for example if it or the
+network fail then something in the environment is responsible for restarting
+the coordinator or for repairing the network, respectively). To fulfil this task the
+coordinator must have access to durable storage for its logs (via a filesystem or in
+a database).  At the time of writing, managing coordinators is the responsibility of
+the user. An "out-of-the-box" solution will be forthcoming.
+
+== Examples
+
+The following is a simple example
+(https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc#java-annotations[taken from the specification document])
+of how to start an LRA and how to receive a notification when the LRA is later
+cancelled (the `@Compensate` annotated method is called) or closed (`@Complete` is called):
+
+[source,java]
+----
+@Path("/")
+@ApplicationScoped
+public class SimpleLRAParticipant
+{
+    @LRA(LRA.Type.REQUIRES_NEW)
+    @Path("/work")
+    @PUT
+    public void doInTransaction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
+    {
+        /*
+         * Perform business actions in the context of the LRA identified by the
+         * value in the injected JAX-RS header. This LRA was started just before
+         * the method was entered (REQUIRES_NEW) and will be closed when the
+         * method finishes at which point the completeWork method below will be
+         * invoked.
```

Comment:
Again for my own understanding. 
You get the request and intercept it. You extract the header and create/enlist the participant into the coordinator. Then, depending on the completion, you notify the coordinator. Am I right?

Reply:
The JAX-RS request filter intercepts incoming JAX-RS requests to see if the target resource contains LRA annotations. If the resource is an LRA participant then it interacts with the coordinator (whether it's remote or in-VM) in order to manage the life-cycle and membership of the LRA. If the coordinator is unavailable it reports the failure back to the caller. The filter interacts with the coordinator using timed async JAX-RS invocations ie it is effectively a blocking call (at the time I wrote it there was no support for async filters and there probably still isn't, although Stephane (Epardaud) did write one that I briefly experimented with).

Reply:
Thanks